CONTRIBUTING.md

```diff
@@ -0,0 +1,125 @@
+# Contributing
+
+## General Workflow
+
+1. Fork the repo
+1. Cut a namespaced feature branch from master
+1. Make commits to your feature branch. Prefix each commit like so:
+1. When you've finished with your fix or feature, merge upstream changes into your branch. submit a pull request directly to master. Include a description of your changes.
+1. Your pull request will be reviewed by another maintainer. The point of code reviews is to help keep the codebase clean and of high quality and, equally as important, to help you grow as a programmer. If your code reviewer requests you make a change you don't understand, ask them why.
+1. Fix any issues raised by your code reviwer, and push your fixes as a single new commit.
```

Comment:
why as a single commit? especially if the review entails changes that touch multiple areas (although it could be argued that PRs shouldn't be this big in the first place)